### PR TITLE
remove goreleaser replacements property

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Goreleaser [deprecated](https://goreleaser.com/deprecations/?h=replacement#archivesreplacements) the replacements yaml property, which has been causing our binary publications to fail for a while now.  This fixes the yaml and introduces a slack message for binary publication failures.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :bug: Bugfix
- [x] :computer: CI/Deployment

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -442,7 +442,8 @@ jobs:
     needs: [Test-Suite-Trusted, Source-Code-Linting, Website-Linting, SetEnv]
     environment: ${{ needs.SetEnv.outputs.environment }}
     runs-on: ubuntu-latest
-    if: startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/main'
+    # TODO: reinstate before merge
+    # if: startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/main'
```
Comment (ashmrtn) [inline .github/workflows/ci.yml]: I'm guessing this TODO still needs addressed? 😉

File: src/.goreleaser.yaml [hunk 2]
```diff
@@ -18,13 +18,15 @@ builds:
...
 archives:
-  - name_template: "{{ .ProjectName }}_{{ .Tag }}_{{ .Os }}_{{ .Arch }}"
-    replacements:
-      darwin: Darwin
-      linux: Linux
-      windows: Windows
-      386: i386
-      amd64: x86_64
+  # this name template makes the OS and Arch compatible with the results of uname.
+  - name_template: >-
```
Comment (ashmrtn) [inline src/.goreleaser.yaml:22]: is this leading line going to behave as expected? I'm not sure if it's GitHub workflow formatting magic or not. It doesn't seem like golang templating DSL

**Review by ashmrtn — APPROVED** [review]

File: src/.goreleaser.yaml (hunk 2 above)
Comment (pandeyabs) [reply src/.goreleaser.yaml:22]: The deprecation notice [seems to propose](https://goreleaser.com/deprecations/?h=replacement#archivesreplacements) this as the alternative to replacements.

Comment (ashmrtn) [reply src/.goreleaser.yaml:22]: the overall text yes, but I'm wondering about the `>-` specifically which I don't think is part of golang's DSL for templates

Comment (pandeyabs) [reply src/.goreleaser.yaml:22]: i see. 

I do see `>-` being used in several other packages. [Example](https://github.com/aquasecurity/trivy/blob/a9badeaba8bb9e908bce2f6f6defecf2f8515aeb/goreleaser.yml#L105)

File: src/.goreleaser.yaml [hunk 3]
```diff
@@ -18,13 +18,15 @@ builds:
...
+      {{ .ProjectName }}_
+      {{- .Tag }}_
+      {{- title .Os }}_
+      {{- if eq .Arch "amd64" }}x86_64
+      {{- else if eq .Arch "386" }}i386
```
Comment (pandeyabs) [inline src/.goreleaser.yaml:27]: We should remove support for 32 bit archs in a future PR, and we should add arm64 support.

File: src/.goreleaser.yaml (hunk 2 above)
Comment (ryanfkeepers) [reply src/.goreleaser.yaml:22]: This came out of goreleasers autogenerated yaml file, so, I take that as a sign that it _must_ work.

File: src/.goreleaser.yaml (hunk 3 above)
Comment (ryanfkeepers) [reply src/.goreleaser.yaml:27]: That's a topic to take up with Georgi and Vaibhav.  I don't recall the original conversations around this, but I assume we retained it intentionally.  Maybe you can make a tech debt ticket for it?

Comment (pandeyabs) [reply src/.goreleaser.yaml:27]: https://github.com/golang/go/wiki/GoArm

**Pull request merged**